components/Image/index.js

```diff
@@ -71,8 +75,19 @@ function Image(props) {
     isPlaceholder
   } = props;
 
+  const treatAsGif = matchesFileExtension(url, 'gif') && !isPlaceholder;
+
+  const gifRenderOps = {
+    format: treatAsGif ? 'mp4' : renderOps && renderOps.format
+  };
+
   // CLEANUP: only apply { resizeMode: crop } if maxSize is specified.
-  const finalRenderOps = Object.assign({}, BASE_RENDER_OPS, renderOps);
+  const finalRenderOps = Object.assign(
+    {},
+    BASE_RENDER_OPS,
+    renderOps,
+    gifRenderOps
```

Comment:
something like this would make it more succinct, i think:
```suggestion
  const finalRenderOps = Object.assign(
    {},
    BASE_RENDER_OPS,
    renderOps,
    ...(treatImageAsVideo && {
      format: 'mp4'
    })
  );
```

you don't need to worry about an actual ternary expression; the `format` is already in `renderOps`. so just spread the `{ format: 'mp4' }` when you need to.

Reply:
yes, classy! I like that much better.